### PR TITLE
Show correct gas density in CCR mode

### PR DESCRIPTION
**Pull request by atdotde**
When collecting the data for the infobox, we have
already computed the current partial pressures of the breathing gas taking into accoutn the divemode. Use those rather than fractions (which for CCR mode are those of diluent) to compute the gas density.

Reported-by: Pietro Tranquillini <p.tranquillini@gmail.com>
Signed-off-by: Robert C. Helling <helling@atdotde.de>

### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### CODE REVIEW

File: core/profile.c [hunk 1]
```diff
@@ -1160,6 +1160,11 @@ static int calculate_ccr_po2(struct plot_data *entry, const struct divecomputer
 	}
 }
 
+double gas_density(const struct gas_pressures *pressures)
```
Comment (bstoeger) [inline core/profile.c]: If this function is used only in this source file, then it should be made of `static` linkage.

I think there is a typo in the commit message: 'diluent' with one 'l'.

Comment (atdotde) [reply core/profile.c]: Addressed your concerns in an updated version.

**Comment by atdotde** [comment]
@dirkhh do you understand why the Android build fails?

**Comment by glance-** [comment]
> Could not GET 'https://jcenter.bintray.com/com/android/tools/build/builder/3.2.1/builder-3.2.1.pom'.
> Read timed out

Clearly external party issues which caused the build to fail.

**Comment by glance-** [comment]
Triggered a re-run of the build, but if we keep seeing such issues we should look at something like:
https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows

**Comment by dirkhh** [comment]
> Triggered a re-run of the build, but if we keep seeing such issues we should look at something like: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows

See my comment elsewhere - I think we need to rev versions in the Android build dependencies again.
Just caching these files isn't really the answer - they usually get removed because there are real bugs in them.

**Review by dirkhh — COMMENTED** [review]
I assume this is correct - this is your code :)

The massive diff in the CSV files is caused by the resulting change to the gas density, I assume?

**Comment by atdotde** [comment]
Indeed, it's slightly different densities that are recorded. I think this is ready to merge.

**Pull request merged**